docs/research/ticket-selection-algorithm.md

```diff
@@ -0,0 +1,8 @@
+# <img class="dcr-icon" src="/img/dcr-icons/QuestionTicket.svg" /> Ticket Selection Algorithm
+
+Every block, 5 tickets are pseudorandomly selected from the ticket pool to [vote](../../proof-of-stake/overview/). Below is a high-level description of the ticket selection algorithm. A more detailed, low-level walkthrough of the ticket selection code can be found in this [video](https://www.youtube.com/watch?v=eysGWVhDFWY).
+
+## Pseudorandom number generation
+
+Tickets are selected from the ticket pool using a deterministic [Pseudorandom Number Generator (PRNG)](https://en.wikipedia.org/wiki/Pseudorandom_number_generator) algorithm. It is deterministic because the seed input to the PRNG algorithm is derived from the hash of the previous block (the block being voted on). Specifically, the seed is a serialization of the [header](../advanced/block-header-specifications.md) of the previous block suffixed with a constant derived from the hex representation of Pi, which acts as a publicly verifyable Nothing-Up-My-Sleeve ([NUMS](https://en.wikipedia.org/wiki/Nothing-up-my-sleeve_number)) constant. All eligible tickets in the ticket pool (also referred to as live tickets) are hashed using the [BLAKE-256](blake-256-hash-function.md) hash function and sorted [lexicographically](https://en.wikipedia.org/wiki/Lexicographical_order) by their hash to generate a [total order](https://en.wikipedia.org/wiki/Total_order), with the total number of eligible tickets as an upper bound. Finally, uniformly random values are produced and used as indicices into the total order.
```

Comment:
The word algorithm shouldn't be after PRNG in the first sentence.  "using a deterministic Pseudorandom Number Generator algorithm" doesn't read properly.  Instead just "using a deterministic Pseudorandom Number Generator" is accurate.

Reply:
Same goes later for PRNG algorithm.  Just PRNG.

Reply:
verifiable

Reply:
Perhaps clarifying Pi here as

> the mathematical constant Pi (𝜋)

would be helpful.

Reply:
> by their hash to generate a total order, with the total number of eligible tickets as an upper bound.

That is not correct since the upper bound doesn't apply to the total order itself rather to the uniformly random values.  How about:

>Finally, uniformly random values constrained by the number of total eligible tickets as an upper bound are produced and used as indices into the total order.

Also note that "indicices" is not spelled correctly in the original text.